.github/workflows/.reusable-docker-build.yml

```diff
@@ -6,30 +6,39 @@ on:
     inputs:
       registry-url:
         type: string
-        description: Github container registry base URL
+        description: Github Container Registry base URL
         required: false
         default: ghcr.io
       file:
         type: string
         description: Path to the Dockerfile
-        required: true
+        required: false
+        default: Dockerfile
+      target:
+        type: string
+        description: Sets the target stage to build
+        required: false
       image-name:
         type: string
-        description: Image slug
+        description: Image name
         required: true
       build-args:
         type: string
         description: List of build-time variables
         required: false
       scan:
         type: boolean
-        description: Whether to scan image for vulnerabilities
+        description: Whether to scan built image for vulnerabilities
         required: false
         default: true
     outputs:
       image:
         description: Resulting image specifier
         value: ${{ inputs.registry-url }}/flagsmith/${{ inputs.image-name }}:${{ jobs.build.outputs.version }}
+    secrets:
+      secrets:
```

Comment:
Perhaps this is some GH workflows oddness but it feels a little strange to define the secrets key twice here?

Reply:
This is a coincidence. Bottom-level `secrets` key exposes the docker build and push action `secrets` argument. Since we're passing Github secrets as values to this argument, we have to use the top-level workflow `secrets` key.